### PR TITLE
feat(INF-70): privatize license-check.yml

### PR DESCRIPTION
**Pull request by welschsn**

### CODE REVIEW

File: .github/workflows/license-check.yml [hunk 1]
```diff
@@ -3,27 +3,10 @@ name: License Check
...
-                     and this file should be used instead of the default one. The download step will be skipped
-                     in that case.
-        required: false
-        type: boolean
-        default: false
-      additionalLicensesURI:
-        description: Location of additional-licenses.yml file, which is optional.
-        required: false
-        type: string
-        default: ''
```
Comment (welschsn) [inline .github/workflows/license-check.yml:22]: these parameters are not used by any project and not necessary

File: .github/workflows/license-check.yml [hunk 2]
```diff
@@ -32,27 +15,41 @@ on:
...
+      jdkVersion:
+        description: 'The JDK version to use. (defaults to "11")'
+        required: false
+        type: string
+        default: '11'
+      javaDistribution:
+        description: 'The Java distribution to use. (defaults to eclipse "temurin")'
+        required: false
+        type: string
+        default: 'temurin'
```
Comment (welschsn) [inline .github/workflows/license-check.yml:27]: making this configurable seems usefull

File: .github/workflows/license-check.yml [hunk 3]
```diff
@@ -65,36 +62,32 @@ jobs:
...
-          -L ${{ inputs.permittedLicensesURI }}
-      - name: Download additional-licenses.yml
-        if: inputs.additionalLicensesURI != ''
+
+      - uses: actions/checkout@v4
+        with:
+          repository: levigo/reusable-workflows
+          ref: main
+          path: ${{ env.REPO_PATH }}
+          token: ${{ secrets.REPO_PTA }}
```
Comment (welschsn) [inline .github/workflows/license-check.yml:71]: we cannot simply download a file from a private repository, so we are cloning the repository with git

File: .github/workflows/license-check.yml [hunk 4]
```diff
@@ -65,36 +62,32 @@ jobs:
...
+
+      - name: Copy license-check files
         run: |
-          curl -H 'Accept: application/vnd.github.v4.raw' \
-          --output-dir ".github" -O \
-          -L ${{ inputs.additionalLicensesURI }}
+          cp ${REPO_PATH}/.github/workflows/license-check/third-party-file_template.ftl .
+          cp ${REPO_PATH}/.github/workflows/license-check/license-check.py .
+          cp ${REPO_PATH}/.github/workflows/license-check/permitted-licenses.yml .github/
+
```
Comment (welschsn) [inline .github/workflows/license-check.yml:78]: copy files from cloned repository to where they are expected by the script

File: .github/workflows/license-check.yml (hunk 3 above)
Comment (benjaminalbanese) [reply .github/workflows/license-check.yml:71]: right, thanks for the explanation.

**Review by benjaminalbanese — APPROVED** [review]

**Pull request merged**